### PR TITLE
Remove onnxruntime/core/protobuf

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

Remove unused files

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by yan12125** [comment]
I remember those are slightly different than corresponding files from ONNX - the former contain ORT-specific stuffs. Things may have changed, too ;)

**Review by souptc — DISMISSED** [review]

**Comment by skottmckay** [comment]
> I remember those are slightly different than corresponding files from ONNX - the former contain ORT-specific stuffs. Things may have changed, too ;)

The only change was this IIRC
```  
message ModelProto {
...
  // kezhan: This field is not in ONNX, and will be pushed into ONNX with good use cases in microsoft.
  repeated FunctionProto functions = 100;
```

I'm not aware of anyone that has actually used that additional field, but I thought the ORT code has some handling of it that may need updating if it hasn't been removed already.

There was also a script or doco somewhere around updating ONNX protobuf defs that referred to using these files which should be updated/removed as well if it still exists.

**Comment by skottmckay** [comment]
> There was also a script or doco somewhere around updating ONNX protobuf defs that referred to using these files which should be updated/removed as well if it still exists.

docs\How_To_Update_ONNX_Dev_Notes.md

**Comment by skottmckay** [comment]
We removed usage of ModelProto.functions in https://github.com/microsoft/onnxruntime/pull/4878 but left some pieces there.
Can we now remove `model_specific_functions` from the Model ctor given the implementation for that ctor completely ignores that argument?

https://github.com/microsoft/onnxruntime/blob/fa722d208b4af6f1c335273a1e465ae0f9d892dc/onnxruntime/core/graph/model.h#L53
https://github.com/microsoft/onnxruntime/blob/fa722d208b4af6f1c335273a1e465ae0f9d892dc/onnxruntime/core/graph/model.cc#L48

I think that would complete the removal.

**Comment by yan12125** [comment]
ONNX 1.10 now also has `repeated FunctionProto functions` in `ModelProto` [1], and there are some updates related to functions in another PR that updates ONNX [2]. Maybe this and that PR need coordination.

[1] https://github.com/onnx/onnx/pull/3532
[2] https://github.com/microsoft/onnxruntime/pull/8458

**Comment by snnn** [comment]
@skottmckay looks like we need to keep these two?

File: docs/How_To_Update_ONNX_Dev_Notes.md [hunk 1]
```diff
@@ -18,15 +18,12 @@ This file should be generated. See [cgmanifests/README](/cgmanifests/README.md)
 3. Update [tools/ci_build/github/linux/docker/scripts/install_onnx.sh](/tools/ci_build/github/linux/docker/scripts/install_onnx.sh).
```
Comment (askhade) [inline docs/How_To_Update_ONNX_Dev_Notes.md:18]: this is not valid any more. Instead we need to update the 2 requirements.txt
https://github.com/microsoft/onnxruntime/blob/master/tools/ci_build/github/linux/docker/scripts/manylinux/requirements.txt
and
https://github.com/microsoft/onnxruntime/blob/master/tools/ci_build/github/linux/docker/scripts/requirements.txt

**Comment by askhade** [comment]
It is OK to remove these local proto files as post PR #4878 ORT has stopped using them.

As both @skottmckay  and @yan12125  pointed out model location functions were only internal to ORT and this was the only difference between proto files in ORT and ONNX. Now we added them in ONNX IR as well in ONNX 1.10 release.

**Comment by askhade** [comment]
> We removed usage of ModelProto.functions in #4878 but left some pieces there.
> Can we now remove `model_specific_functions` from the Model ctor given the implementation for that ctor completely ignores that argument?
> 
> https://github.com/microsoft/onnxruntime/blob/fa722d208b4af6f1c335273a1e465ae0f9d892dc/onnxruntime/core/graph/model.h#L53
> 
> 
> https://github.com/microsoft/onnxruntime/blob/fa722d208b4af6f1c335273a1e465ae0f9d892dc/onnxruntime/core/graph/model.cc#L48
> 
> I think that would complete the removal.

I am working on a PR to support model local functions again in ORT. My PR can take care of this.

**Comment by snnn** [comment]
I will update the docs.

**Review by skottmckay — APPROVED** [review]
:shipit:

**Pull request merged**